### DISCUSSION TITLE
using-oidc-with-terraform-in-github-actions/

### DISCUSSION THREAD

**Discussion by giscus** [colindembovsky/colindembovsky.github.io#22] [Giscussions]
# using-oidc-with-terraform-in-github-actions/

I’ve posted before about how to authenticate to Azure in GitHub Actions using OIDC. It should follow that Terraform templates would be easy to use - but there are some gotchas.

https://colinsalmcorner.com/using-oidc-with-terraform-in-github-actions/

**Comment by thegooddalton** [comment]
I've done pretty much the same thing as you here, except my terraform plan keeps failing and i cannot figure out why!? Did you run into anything like this..

ValidateProviderConfig from "provider[\"registry.terraform.io/hashicorp/azurerm\"]" changed the config value, but that value is unused

**Reply by thegooddalton** [reply]
Appears to be a provider issue, downgrading from 3.70.0 to 3.7.0 seems to have done the trick...

**Reply by thegooddalton** [reply]
Pls bump if you can!

https://github.com/hashicorp/terraform-provider-azurerm/issues/23766
